### ISSUE TITLE
Provide a quick fix for 'module does not refer to a type'

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/TypeScript#24172]
```ts
// ./foo.ts

export async function bar() {
    let foo: import("./foo") = await import("./foo");
//           ^^^^^^^^^^^^^^^
// Module './foo' does not refer to a type, but is used as a type here.
}
```

We should give a quick fix in this scenario.

**Comment by mhegazy** [comment]
//CC @Kingwl

**Comment by mhegazy** [comment]
change it to `typeof import("./foo")` if applicable.

**Comment by Kingwl** [comment]
```ts
// @Filename: ./foo.ts
//// interface Point {
////     x: number;
////     y: number;
//// }
//// export = Point;

// @Filename: ./bar.ts
//// const a: import("./foo") = import("./foo")

```

need some help 
in this case, there are no Diagnostics emitted 
but every thing is ok on 
```ts
//// declare module "foo" {
////     const a = "foo"
////     export = a
//// }
//// const x: import("foo") = "foo";

```

have i missing something?
